src/test/java/com/stripe/functional/PlanTest.java

```diff
@@ -15,101 +18,36 @@
-    public void testCustomerCreateWithPlan() throws StripeException {
-        Plan plan = Plan.create(getUniquePlanParams());
-        Customer customer = createDefaultCustomerWithPlan(plan);
-        assertEquals(customer.getSubscriptions().getData().get(0).getPlan().getId(), plan.getId());
-    }
-
-    @Test
-    public void testPlanCreatePerCallAPIKey() throws StripeException {
-        Plan plan = Plan.create(getUniquePlanParams(), Stripe.apiKey);
-        assertEquals(plan.getInterval(), "month");
-    }
-
-    @Test
-    public void testPlanUpdatePerCallAPIKey() throws StripeException {
-        Plan createdPlan = Plan.create(getUniquePlanParams(), Stripe.apiKey);
-        Map<String, Object> updateParams = new HashMap<String, Object>();
-        updateParams.put("name", "Updated Plan Name");
-        Plan updatedplan = createdPlan.update(updateParams, Stripe.apiKey);
-        assertEquals(updatedplan.getName(), "Updated Plan Name");
-    }
-
-    @Test
-    public void testPlanRetrievePerCallAPIKey() throws StripeException {
-        Plan createdPlan = Plan.create(getUniquePlanParams(), Stripe.apiKey);
-        Plan retrievedPlan = Plan.retrieve(createdPlan.getId(), Stripe.apiKey);
-        assertEquals(createdPlan.getId(), retrievedPlan.getId());
-    }
-
-    @Test
-    public void testPlanListPerCallAPIKey() throws StripeException {
-        Map<String, Object> listParams = new HashMap<String, Object>();
-        listParams.put("count", 1);
-        List<Plan> Plans = Plan.all(listParams, Stripe.apiKey).getData();
-        assertEquals(Plans.size(), 1);
-    }
-
-    @Test
-    public void testPlanDeletePerCallAPIKey() throws StripeException {
-        Plan createdPlan = Plan.create(getUniquePlanParams(), Stripe.apiKey);
-        DeletedPlan deletedPlan = createdPlan.delete(Stripe.apiKey);
-        assertTrue(deletedPlan.getDeleted());
-        assertEquals(deletedPlan.getId(), createdPlan.getId());
-    }
-
-    @Test
-    public void testPlanMetadata() throws StripeException {
-        testMetadata(Plan.create(getUniquePlanParams()));
-    }
+public class PlanTest extends BaseStripeStubTest {
+	@Test
+	public void testPlanCreate() throws StripeException {
+		Map<String, Object> params = new HashMap<String, Object>();
+		params.put("id", "gold");
+		Plan _plan = Plan.create(params);
+	}
+
+	@Test
+	public void testPlanRetrieve() throws StripeException {
+		Plan _plan = Plan.retrieve("gold");
+	}
+
+	@Test
+	public void testPlanUpdate() throws StripeException {
+		Plan plan = Plan.retrieve("gold");
+		Map<String, Object> params = new HashMap<String, Object>();
+		params.put("name", "silver");
+		Plan _plan = plan.update(params);
+	}
+
+	@Test
+	public void testPlanDelete() throws StripeException {
+		Plan plan = Plan.retrieve("gold");
+		DeletedPlan _deletedPlan = plan.delete();
+	}
+
+	@Test
+	public void testPlanList() throws StripeException {
+		PlanCollection plans = Plan.all(null);
+		assertEquals(1, plans.getData().size());
+	}
```

Comment:
Functional tests change to just a simple test for ever supported API operation on a resource. Response data is generated anyway, so we don't assert strongly against it. Instead we just check that it's the basic format that we expected which tells us that we called the right URL and were able to deserialize the response.